plugins/bufferlines/bufferline.nix

```diff
@@ -125,196 +215,395 @@ in
             ```
           '';
 
-      bufferCloseIcon = helpers.defaultNullOpts.mkStr "" "The close icon for each buffer.";
+      buffer_close_icon = defaultNullOpts.mkStr "" ''
+        The close icon for each buffer.
+      '';
+
+      modified_icon = defaultNullOpts.mkStr "●" ''
+        The icon indicating a buffer was modified.
+      '';
 
-      modifiedIcon = helpers.defaultNullOpts.mkStr "●" "The icon indicating a buffer was modified.";
+      close_icon = defaultNullOpts.mkStr "" "The close icon.";
 
-      closeIcon = helpers.defaultNullOpts.mkStr "" "The close icon.";
+      close_command = defaultNullOpts.mkStr "bdelete! %d" ''
+        Command or function run when closing a buffer.
+      '';
 
-      closeCommand = helpers.defaultNullOpts.mkStr "bdelete! %d" "Command or function run when closing a buffer.";
+      custom_filter = defaultNullOpts.mkLuaFn null ''
+        ```
+        fun(buf: number, bufnums: number[]): boolean
 
-      leftMouseCommand = helpers.defaultNullOpts.mkStr "buffer %d" "Command or function run when clicking on a buffer.";
+        NOTE: this will be called a lot so don't do any heavy processing here.
+        ```
+      '';
 
-      rightMouseCommand = helpers.defaultNullOpts.mkStr "bdelete! %d" "Command or function run when right clicking on a buffer.";
+      left_mouse_command = defaultNullOpts.mkStr "buffer %d" ''
+        Command or function run when clicking on a buffer.
+      '';
 
-      middleMouseCommand = helpers.defaultNullOpts.mkStr null "Command or function run when middle clicking on a buffer.";
+      right_mouse_command = defaultNullOpts.mkStr "bdelete! %d" ''
+        Command or function run when right clicking on a buffer.
+      '';
+
+      middle_mouse_command = defaultNullOpts.mkStr null ''
+        Command or function run when middle clicking on a buffer.
+      '';
 
       indicator = {
-        icon = helpers.defaultNullOpts.mkStr "▎" "icon";
+        icon = defaultNullOpts.mkStr "▎" ''
+          Indicator icon.
 
-        style = helpers.defaultNullOpts.mkEnumFirstDefault [
-          "icon"
-          "underline"
-        ] "style";
+          This should be omitted if indicator style is not `icon`.
+        '';
+
+        style =
+          defaultNullOpts.mkEnumFirstDefault
+            [
+              "icon"
+              "underline"
+              "none"
+            ]
+            ''
+              Indicator style.
+            '';
```

Comment:
Yeah, I just started doing it all indented strings because of nits in the previous PR. I think for these blocks I agree it makes sense to be a short string for formatting. Some of the individual options is does seem easier to read with indented strings.